src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala

```diff
@@ -90,6 +90,30 @@ final case class TenantApiServiceImpl(
     }
   }
 
+  override def getConsumers(name: Option[String], offset: Int, limit: Int)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerTenants: ToEntityMarshaller[Tenants]
+  ): Route = {
+    implicit val problemMarshaller: ToEntityMarshaller[Problem] = entityMarshallerProblem
+    authorize(ADMIN_ROLE, API_ROLE, SECURITY_ROLE) {
+      logger.info(s"Retrieving Consumers with name = $name, limit = $limit, offset = $offset")
+
+      val result: Future[Tenants] = for {
+        requesterId <- getOrganizationIdFutureUUID(contexts)
```

Comment:
the requirement is to filter consumers having agreements for requester's eservices.
I saw mainly 2 alternatives:
1. make the producer id a parameter of this endpoint, and let the BFF extract the organization from the jwt and pass it here
2. extract the requesterId (producerId) directly in this endpoint - the one implemented

Solution 1 allows to have a more reusable endpoint, but is up to the caller to decide which producer is used in filter.
I saw solution 2 as an additional authorization check, but if you think solution 1 is better I can move to that.

A penny for your thoughts

Reply:
If the scope of this endpoint is restricted only to the requester then I agree with you: solution 2 is my choice too.

Reply:
My feeling about it might not be so neatly defined as yours, but yes, I prefer solution 2 too.